### PR TITLE
[V1] Add logits_processors support

### PR DESCRIPTION
**Pull request by imkero**
Add `SamplingParams.logits_processors` support for V1

### CODE REVIEW

File: vllm/v1/sample/sampler.py [hunk 1]
```diff
@@ -134,3 +137,30 @@ def apply_penalties(
...
+
+    def apply_logits_processors(
+        self,
+        logits: torch.Tensor,
+        sampling_metadata: SamplingMetadata,
+    ) -> torch.Tensor:
+        for seq_idx, seq_logits_processors in \
+            enumerate(sampling_metadata.logits_processors):
+            if not seq_logits_processors:
+                continue
```
Comment (mgoin) [inline vllm/v1/sample/sampler.py:149]: It seems like a mandatory performance loss that we will always iterate through this list `max_num_reqs` times, even if there are no logit processors. Would it be worse to use a dynamically allocated Dict or structure that only includes the seqs that have logit processors?

Comment (imkero) [reply vllm/v1/sample/sampler.py:149]: I think it's ok to use a dict here

**Comment by imkero** [comment]
Maybe still some overhead near `parameters = inspect.signature(logits_processor).parameters`, trying to enhance it

**Comment by WoosukKwon** [comment]
+1 for @njhill's comment
